### PR TITLE
Cleanup plot types

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Some simplifications and more standard code style.

### CODE REVIEW

File: plot_types/basic/scatter_plot.py [hunk 1]
```diff
@@ -12,15 +12,15 @@
 
 # make the data
 np.random.seed(3)
-X = 4 + np.random.normal(0, 2, 24)
-Y = 4 + np.random.normal(0, 2, len(X))
+x = 4 + np.random.normal(0, 2, 24)
+y = 4 + np.random.normal(0, 2, len(x))
 # size and color:
-S = np.random.uniform(15, 80, len(X))
+S = np.random.uniform(15, 80, len(x))
```
Comment (QuLogic) [inline plot_types/basic/scatter_plot.py:18]: Why is this not lowercased? Or maybe rename to `sizes`.

File: plot_types/stats/barbs.py [hunk 2]
```diff
@@ -14,14 +14,13 @@
 np.random.seed(1)
 X = [[2, 4, 6]]
 Y = [[1.5, 3, 2]]
-U = -np.ones((1, 3)) * 0
-V = -np.ones((1, 3)) * np.linspace(50, 100, 3)
+U = np.zeros_like(X)
+V = -np.ones_like(X) * np.linspace(50, 100, 3)
```
Comment (QuLogic) [inline plot_types/stats/barbs.py:18]: Lower case?

File: plot_types/stats/barbs.py [hunk 3]
```diff
@@ -14,14 +14,13 @@
 np.random.seed(1)
 X = [[2, 4, 6]]
 Y = [[1.5, 3, 2]]
-U = -np.ones((1, 3)) * 0
-V = -np.ones((1, 3)) * np.linspace(50, 100, 3)
+U = np.zeros_like(X)
```
Comment (story645) [inline plot_types/stats/barbs.py:17]: This was made lowercase everywhere else. Why uppercase here?

**Review by story645 — APPROVED** [review]

File: plot_types/stats/barbs.py (hunk 3 above)
Comment (timhoffm) [reply plot_types/stats/barbs.py:17]: I mostly follow the capitalization in the signature of the respective plot function, some of which are lowercase, some are uppercase.

There's a loose convention that 1D arrays have regular lowercase names and 2D arrays from `np.meshgrid()` have uppercase names (e.g. `X, Y = np.meshgrid(x, y)`). I suspect that the signature roughly follows this pattern: If we expect 2D input, we use an uppercase parameter name. For `barbs` this is a bit ambiguous as parameters can be 1D or 2D, so I followed the signature style.

File: plot_types/stats/barbs.py (hunk 2 above)
Comment (timhoffm) [reply plot_types/stats/barbs.py:18]: See comment in https://github.com/matplotlib/matplotlib/pull/20509#pullrequestreview-692294629

**Comment by QuLogic** [comment]
I'm not sure that message is what you meant to put in that last commit.

**Comment by timhoffm** [comment]
Squashed.

**Pull request merged**